### ISSUE TITLE
[deploy] Deploy to Sonatype Nexus: 502 bad gateway

### ISSUE THREAD

**Issue by TomCools** [jreleaser/jreleaser#1058]
Dear maintainer of the project, I hope you are having a nice day! Thank you for all the hard work you are putting into this project. <3

I do however, have a bug ;) I'm not in a big rush, but it is breaking a new feature, so here goes. If you need more information, leave a comment and I'll get back to you.

### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Correctly stage artifacts (https://jreleaser.org/guide/latest/examples/maven/staging-artifacts.html)
2. Correctly setup the properties 

- JRELEASER_NEXUS2_MAVEN_CENTRAL_USERNAME="EXCLUDED"
- JRELEASER_NEXUS2_MAVEN_CENTRAL_PASSWORD="EXCLUDED"
- JRELEASER_GPG_SECRET_KEY="EXCLUDED"
- JRELEASER_GPG_PUBLIC_KEY="EXCLUDED"

3. Setup Maven Plugin
See project POM: https://github.com/TomCools/rickroll-security-spring-boot-starter/blob/jreleaser/pom.xml

4. Run jreleaser:deploy

### Expected Behaviour

I expect that when I use JReleaser Nexus 2 deployment, that it is uploading all my assets to my nexus repository.

### Actual Behaviour

I get a 502 Bad Gateway (see trace.log below).

**What I've checked:**
- The staging repository gets created, so it's not an authentication issue;
- I was able to upload to the Nexus using the _nexus-staging-maven-plugin_. So my GPG keys and such should be fine;
- I already looked at the JReleaser code. It does a PUT request to upload the files. I mimic'd that behavior with Postman and the upload does succeed, so it's not an issue at Sonatype Nexus. 

![image](https://user-images.githubusercontent.com/11362563/202925561-ed650727-3b4a-4eab-b727-c8e513d87521.png)

![image](https://user-images.githubusercontent.com/11362563/202925579-77aa73da-2bad-46b7-9dd5-05d8cfbd9822.png)

### Environment Information

- **Operating System**: windows-x86_64
- **JReleaser Version:** 1.3.1
- **JDK Version:** 18.0.1

## Stacktrace

```
[INFO]  Project version set to 3.0.0-RC1
[INFO]  Release is not snapshot
[INFO]  Timestamp is 2022-11-20T21:39:20.5151269+01:00
[INFO]  HEAD is at 2edd500
[INFO]  Platform is windows-x86_64
[DEBUG] loading extension default-jreleaser-mustache:org.jreleaser.extensions.internal.mustache.DefaultMustacheExtension
[DEBUG] adding org.jreleaser.extensions.internal.mustache.DefaultMustacheExtensionPoint from default-jreleaser-mustache extension
[INFO]  dry-run set to false
[INFO]  Deploying all staged artifacts
[INFO]    [nexus2] Deploying to maven-central
[DEBUG]   [nexus2] Cannot run program "pomchecker.bat": CreateProcess error=2, The system cannot find the file specified
[DEBUG]   [nexus2] cached C:\Users\tomco\.jreleaser\caches\pomchecker\1.3.0\pomchecker-1.3.0\bin\pomchecker.bat
[ERROR]   [nexus2] 
[DEBUG]   [nexus2] Command execution error. exitValue = 9009
[WARN]    [nexus2] Could not verify pomchecker 1.3.0. Skipping
[WARN]    [nexus2] pomchecker is not available. Skipping
[INFO]    [nexus2] Lookup staging profile for be.tomcools
[INFO]    [nexus2] Creating a staging repository for be.tomcools
[DEBUG]   [nexus2] creating staging repository for be.tomcools:12604c09fe59a
[INFO]    [nexus2]  - rickroll-security-spring-boot-starter-3.0.0-RC1-javadoc.jar
[DEBUG]   [nexus2]  - deploying rickroll-security-spring-boot-starter-3.0.0-RC1-javadoc.jar as \be\tomcools\rickroll-security-spring-boot-starter\3.0.0-RC1/rickroll-security-spring-boot-starter-3.0.0-RC1-javadoc.jar
[DEBUG]   [nexus2] url: https://oss.sonatype.org/service/local/staging/deployByRepositoryId/betomcools-1046\be\tomcools\rickroll-security-spring-boot-starter\3.0.0-RC1/rickroll-security-spring-boot-starter-3.0.0-RC1-javadoc.jar
[DEBUG]   [nexus2] opening connection
[DEBUG]   [nexus2] configuring connection
[DEBUG]   [nexus2] Accept: [*/*]
[DEBUG]   [nexus2] Expect: [100-continue]
[DEBUG]   [nexus2] User-Agent: [JReleaser/1.3.1]
[DEBUG]   [nexus2] Content-Type: [application/java-archive]
[DEBUG]   [nexus2] sending data
[DEBUG]   [nexus2] handling response
[ERROR]   [nexus2]  x rickroll-security-spring-boot-starter-3.0.0-RC1-javadoc.jar
org.jreleaser.sdk.nexus2.Nexus2Exception: Error when deploying artifact rickroll-security-spring-boot-starter-3.0.0-RC1-javadoc.jar
	at org.jreleaser.sdk.nexus2.Nexus2.fail(Nexus2.java:251)
	at org.jreleaser.sdk.nexus2.Nexus2.deploy(Nexus2.java:242)
	at org.jreleaser.sdk.nexus2.Nexus2MavenDeployer.deploy(Nexus2MavenDeployer.java:99)
	at org.jreleaser.engine.deploy.maven.ProjectMavenDeployer.deploy(ProjectMavenDeployer.java:55)
	at org.jreleaser.engine.deploy.maven.MavenDeployers.deploy(MavenDeployers.java:124)
	at org.jreleaser.engine.deploy.maven.MavenDeployers.deploy(MavenDeployers.java:111)
	at org.jreleaser.workflow.DeployWorkflowItem.doInvoke(DeployWorkflowItem.java:35)
	at org.jreleaser.workflow.AbstractWorkflowItem.lambda$invoke$0(AbstractWorkflowItem.java:43)
	at org.jreleaser.engine.hooks.CommandHookExecutor.execute(CommandHookExecutor.java:52)
	at org.jreleaser.workflow.AbstractWorkflowItem.invoke(AbstractWorkflowItem.java:43)
	at org.jreleaser.workflow.WorkflowImpl.execute(WorkflowImpl.java:102)
	at org.jreleaser.maven.plugin.JReleaserDeployMojo.execute(JReleaserDeployMojo.java:78)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:577)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:47)
Caused by: org.jreleaser.model.spi.upload.UploadException: Got 502 reason: Bad Gateway, <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>502 Bad Gateway</title>
</head><body>
<h1>Bad Gateway</h1>
<p>The proxy server received an invalid
response from an upstream server.<br />
</p>
</body></html>

	at org.jreleaser.sdk.commons.ClientUtils.uploadFile(ClientUtils.java:272)
	at org.jreleaser.sdk.commons.ClientUtils.putFile(ClientUtils.java:206)
	at org.jreleaser.sdk.nexus2.Nexus2.deploy(Nexus2.java:234)
	... 31 more
[INFO]  Writing output properties to target\jreleaser\output.properties
[ERROR] JReleaser failed after 10,402 s
org.jreleaser.model.JReleaserException: Unexpected error
	at org.jreleaser.engine.deploy.maven.MavenDeployers.deploy(MavenDeployers.java:128)
	at org.jreleaser.engine.deploy.maven.MavenDeployers.deploy(MavenDeployers.java:111)
	at org.jreleaser.workflow.DeployWorkflowItem.doInvoke(DeployWorkflowItem.java:35)
	at org.jreleaser.workflow.AbstractWorkflowItem.lambda$invoke$0(AbstractWorkflowItem.java:43)
	at org.jreleaser.engine.hooks.CommandHookExecutor.execute(CommandHookExecutor.java:52)
	at org.jreleaser.workflow.AbstractWorkflowItem.invoke(AbstractWorkflowItem.java:43)
	at org.jreleaser.workflow.WorkflowImpl.execute(WorkflowImpl.java:102)
	at org.jreleaser.maven.plugin.JReleaserDeployMojo.execute(JReleaserDeployMojo.java:78)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:577)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:47)
Caused by: org.jreleaser.model.spi.deploy.DeployException: Unexpected error when deploying target\staging-deploy\be\tomcools\rickroll-security-spring-boot-starter\3.0.0-RC1\rickroll-security-spring-boot-starter-3.0.0-RC1-javadoc.jar
	at org.jreleaser.sdk.nexus2.Nexus2MavenDeployer.deploy(Nexus2MavenDeployer.java:102)
	at org.jreleaser.engine.deploy.maven.ProjectMavenDeployer.deploy(ProjectMavenDeployer.java:55)
	at org.jreleaser.engine.deploy.maven.MavenDeployers.deploy(MavenDeployers.java:124)
	... 28 more
Caused by: org.jreleaser.sdk.nexus2.Nexus2Exception: Error when deploying artifact rickroll-security-spring-boot-starter-3.0.0-RC1-javadoc.jar
	at org.jreleaser.sdk.nexus2.Nexus2.fail(Nexus2.java:251)
	at org.jreleaser.sdk.nexus2.Nexus2.deploy(Nexus2.java:242)
	at org.jreleaser.sdk.nexus2.Nexus2MavenDeployer.deploy(Nexus2MavenDeployer.java:99)
	... 30 more
Caused by: org.jreleaser.model.spi.upload.UploadException: Got 502 reason: Bad Gateway, <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>502 Bad Gateway</title>
</head><body>
<h1>Bad Gateway</h1>
<p>The proxy server received an invalid
response from an upstream server.<br />
</p>
</body></html>

	at org.jreleaser.sdk.commons.ClientUtils.uploadFile(ClientUtils.java:272)
	at org.jreleaser.sdk.commons.ClientUtils.putFile(ClientUtils.java:206)
	at org.jreleaser.sdk.nexus2.Nexus2.deploy(Nexus2.java:234)
	... 31 more
```

**Comment by aalmiray** [comment]
Tom, do you happen to have a proxy or firewall in place? Perhaps a vpn connection?

**Comment by aalmiray** [comment]
On a second look, the url looks weird because it has a mix of forward and backslash. That might be the problem.

**Comment by TomCools** [comment]
This is on my home network, without VPN or Proxy. If that was an issue, i'd expect the Postman call and the other nexus maven plugin to also fail.

In regards to the slashes in the URL, that works in Postman. Might be an issue in the Java code then. 

I am willing to investigate this myself btw. 😊 i got everything setup already.

(I had sent the above by mail reply... but github didn't show it).

**Comment by aalmiray** [comment]
Thanks. I'm hoping this is a path issue as I have no way to test on Windows.

**Comment by TomCools** [comment]
I'll report my finding when i get to it. (Wednesday at the earliest). I do have a Windows machine to test this on. 👍

**Comment by aalmiray** [comment]
Thank you @TomCools! It would be great if you can test by building your own snapshot of the jereleaser-maven-plugin.

**Comment by aalmiray** [comment]
Released in v1.4.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.4.0
